mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/gridfs/GridFsTest.java

```diff
@@ -0,0 +1,197 @@
+		em.getTransaction().commit();
+	}
+
+	@Test
+	public void canReplaceBlobInEntity() throws SQLException {
+		em.getTransaction().begin();
+		Photo photo = em.find( Photo.class, ENTITY_ID_1 );
+		assertThat( photo ).isNotNull();
+		assertThat( photo.getContent() ).isNotNull();
+		Blob content = photo.getContent();
+		assertThat( content.length() ).isEqualTo( BLOB_CONTENT_1.length );
+		assertThat( content.getBytes( 1, BLOB_CONTENT_1.length ) ).isEqualTo( BLOB_CONTENT_1 );
+
+		Blob blob2 = Hibernate.getLobCreator( em.unwrap( Session.class ) ).createBlob( BLOB_CONTENT_2 );
+		photo.setContent( blob2 );
+		em.merge( photo );
+		em.getTransaction().commit();
+
+		em.getTransaction().begin();
+
+		photo = em.find( Photo.class, ENTITY_ID_1 );
+		assertThat( photo ).isNotNull();
+		assertThat( photo.getContent() ).isNotNull();
+		content = photo.getContent();
+		assertThat( content.length() ).isEqualTo( BLOB_CONTENT_2.length );
+		assertThat( content.getBytes( 1, BLOB_CONTENT_2.length ) ).isEqualTo( BLOB_CONTENT_2 );
+		em.getTransaction().commit();
+
+		GridFSBucket gridFSFilesBucket = GridFSBuckets.create( mongoDatabase, BUCKET_NAME );
+		MongoCursor<GridFSFile> cursor = gridFSFilesBucket.find().iterator();
+		List<GridFSFile> files = new LinkedList<>(  );
+		for ( ; cursor.hasNext(); ) {
+			files.add( cursor.next() );
+		}
+		assertThat( files.size() ).isEqualTo( 1 );
+	}
+
+	@Test
+	public void canRemoveEntityWithBlob() throws SQLException {
+
+		em.getTransaction().begin();
+		Photo photo = new Photo();
+		photo.setId( ENTITY_ID_2 );
+		photo.setDescription( "photo2" );
+		Blob blob = Hibernate.getLobCreator( em.unwrap( Session.class ) ).createBlob( BLOB_CONTENT_1 );
+		photo.setContent( blob );
+		em.persist( photo );
+		em.getTransaction().commit();
+
+		em.getTransaction().begin();
+		photo = em.find( Photo.class, ENTITY_ID_2 );
+		assertThat( photo ).isNotNull();
+		assertThat( photo.getContent() ).isNotNull();
+		Blob content = photo.getContent();
+		assertThat( content.length() ).isEqualTo( BLOB_CONTENT_1.length );
+		assertThat( content.getBytes( 1, BLOB_CONTENT_1.length ) ).isEqualTo( BLOB_CONTENT_1 );
+		em.remove( photo );
+		em.getTransaction().commit();
+
+		GridFSBucket gridFSFilesBucket = GridFSBuckets.create( mongoDatabase, BUCKET_NAME );
+		MongoCursor<GridFSFile> cursor = gridFSFilesBucket.find().iterator();
+		List<GridFSFile> files = new LinkedList<>(  );
+		for ( ; cursor.hasNext(); ) {
+			files.add( cursor.next() );
+		}
+		assertThat( files.size() ).isEqualTo( 1 );
+	}
+
+
+	private MongoDatabase getCurrentDB(EntityManager em) {
+		Session session = em.unwrap( Session.class );
+		OgmSessionFactoryImpl sessionFactory = (OgmSessionFactoryImpl) session.getSessionFactory();
+		OgmDialect dialect = (OgmDialect) sessionFactory.getDialect();
+		GridDialect gridDialect = dialect.getGridDialect();
+		MongoDBDialect mongoDBDialect = null;
+		if ( gridDialect instanceof MongoDBDialect ) {
```

Comment:
THe dialect will always be the MongoDBDialect because this test is specific for mongodb.
You can get the MongoDBDialect with the following code:
```
MongoDBDialect mongoDbDialect = (MongoDBDialect) getServiceRegistry().getService(GridDialect.class);

```